common/winobjc.versioning.common.targets

```diff
@@ -78,9 +66,47 @@
       <PackageVersion_Major>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['MAJOR'].Value)</PackageVersion_Major>
       <PackageVersion_Minor>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['MINOR'].Value)</PackageVersion_Minor>
       <PackageVersion_Patch>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['PATCH'].Value)</PackageVersion_Patch>
-      <PackageVersion_PreReleaseLabel>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['LABEL'].Value)</PackageVersion_PreReleaseLabel>
+      <PackageVersion_PreRelease>$([System.Text.RegularExpressions.Regex]::Match($(PackageVersionOverride), $(_PackageVersionRegex)).Groups['LABEL'].Value)</PackageVersion_PreRelease>
+    </PropertyGroup>
+
+    <!-- Common properties needed to either check the build numbner or to calculate the version -->
+    <PropertyGroup Condition="'$(PackageVersionOverride)' == ''">
+      <PackageVersion_Major>$(GitVersion_Major)</PackageVersion_Major>
+      <PackageVersion_Minor>$(GitVersion_Minor)</PackageVersion_Minor>
+      
+      <PackageVersion_Patch Condition="'$(PackageVersion_Patch)' == '' And '$(GitVersion_PreReleaseLabel)' == ''">$([System.DateTime]::Now.ToUniversalTime().ToString(yyMMdd))</PackageVersion_Patch>
+      <PackageVersion_Patch Condition="'$(PackageVersion_Patch)' == '' And '$(GitVersion_PreReleaseLabel)' != ''">$(GitVersion_Patch)</PackageVersion_Patch>
+
+      <PackageVersion_PreReleaseLabel>$(GitVersion_PreReleaseLabel)</PackageVersion_PreReleaseLabel>
+    </PropertyGroup>
+
+    <!-- Last up, use calculated values. -->
+    <PropertyGroup Condition="'$(PackageVersionOverride)' == ''">
+      <!-- 
+         Three places to get a build timestamp: 
+         1. Off of the build number for a build server build (Before its been modified to be the package version)
```

Comment:
That wasn't a specific goal of this change but yes. After this it will choose to prefer the build number over the timestamp file so it will use a uniform version across solutions

---
In reply to: [113305792](https://github.com/Microsoft/WinObjC/pull/2606#discussion_r113305792) [](ancestors = 113305792)